node/src/asset_cleanup.rs

```diff
@@ -0,0 +1,416 @@
+    current_epoch_id: EpochId,
+) -> AssetCleanup {
+    tracing::info!("Updating from legacy node");
+    if previous_epoch_id != current_epoch_id {
+        tracing::info!(
+            "We should not be updating during a resharing, but we will try our best. Current epoch id: {:?}, old epoch id {:?}.",
+            current_epoch_id, previous_epoch_id
+        );
+        AssetCleanup::DeleteAll
+    } else {
+        AssetCleanup::Keep
+    }
+}
+
+// This function compares the `current_epoch_data` with the epoch data stored in the database.
+// If the epoch id changed, then all assets (triples & presignatures) are deleted.
+// If the epoch id did not change, then the node will discard all owned assets (triples &
+// presignatures) involving any nodes that changed their TLS key.
+pub fn delete_stale_triples_and_presignatures(
+    db: &Arc<SecretDB>,
+    current_epoch_data: EpochData,
+    my_participant_id: primitives::ParticipantId,
+    ecdsa_domain_ds: Vec<DomainId>,
+) -> anyhow::Result<()> {
+    let current_epoch_id = current_epoch_data.epoch_id;
+    let asset_cleanup: AssetCleanup = match get_epoch_data(db)? {
+        None => AssetCleanup::Keep,
+        Some(previous_epoch_data) => match previous_epoch_data {
+            EpochDataWrapper::Legacy(previous_epoch_id) => {
+                get_cleanup_behavior_during_update(previous_epoch_id, current_epoch_id)
+            }
+            EpochDataWrapper::Current(previous_epoch_data) => {
+                get_cleanup_behavior(&previous_epoch_data, &current_epoch_data)
+            }
+        },
+    };
+
+    match asset_cleanup {
+        AssetCleanup::Keep => {}
+        AssetCleanup::DeleteAll => {
+            let mut update_writer = db.update();
+            let _ = update_writer.delete_all(DBCol::Presignature);
+            let _ = update_writer.delete_all(DBCol::Triple);
+            update_writer.commit()?;
+        }
+        AssetCleanup::KeepOnly(persitent_participants) => {
+            // cleanup triples:
+            clean_db::<PairedTriple>(
+                db,
+                DBCol::Triple,
+                &persitent_participants,
+                my_participant_id,
+                TRIPLE_STORE_DOMAIN_ID,
+            )?;
+            // cleanup presignatures:
+            for domain_id in &ecdsa_domain_ds {
+                clean_db::<PresignOutputWithParticipants>(
+                    db,
+                    DBCol::Presignature,
+                    &persitent_participants,
+                    my_participant_id,
+                    Some(*domain_id),
+                )?;
+            }
+        }
+    }
+
+    let mut update_writer = db.update();
+    tracing::info!("Updating epoch data: {:?}.", current_epoch_data);
+    let bytes = bincode::serialize(&current_epoch_data)?;
+    update_writer.put(DBCol::EpochData, EPOCH_ID_KEY, &bytes);
+    tracing::info!("Updated epoch id entry");
+    update_writer.commit()?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod test_utils {
```

Comment:
This is quite a lot of logic. I'd prefer to move it out to a separate file.

Reply:
Yeah, I was wondering about the right place for the logic in this file. I see the following optionss.
- make a `db` folder and move the logic from `db.rs` to that folder, together with the one in this file
- make an `assets` folder and move this logic there
- keep this in the `src` folder and just make yet another file behind the `test` compilation flag.

Reply:
From my shallow knowledge, I'd start with just replacing this module with an `asset_cleanup/test_utils.rs` module because that's what we have right now.

However, pondering some more it boils down to the purpose of the module. Is this general test utilities for database interactions? -> `db/test_utils.rs`, or is this relating only to asset generation then `assets/test_utils.rs` makes more sense.

Reply:
Pondering some more about this I'd propose to make `assets_cleanup` as submodule of `assets` as well. What would you think about this structure for the modules?

```
assets.rs
assets/cleanup.rs
assets/test_utils.rs
```

Reply:
done, nice!